addons/binding/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/handler/KonnectedHandler.java

```diff
@@ -0,0 +1,476 @@
+     */
+    public KonnectedHandler(Thing thing, KonnectedHTTPServlet webHookServlet, String hostAddress, String port) {
+        super(thing);
+        this.webHookServlet = webHookServlet;
+        callbackIpAddress = hostAddress + ":" + port;
+        logger.debug("The callback ip address is: {}", callbackIpAddress);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // get the zone number in integer form
+        Integer zone = Integer.parseInt(channelUID.getId().substring((channelUID.getId().length() - 1)));
+        logger.debug("The channelUID is: {} and the zone is :", channelUID.getAsString(), zone);
+        // convert the zone to the pin based on value at index of zone
+        Integer pin = Arrays.asList(PIN_TO_ZONE).get(zone);
+        logger.debug("Zachary test {}", command instanceof OnOffType);
+        // if the command is OnOfftype
+        if (command instanceof OnOffType) {
+            int sendCommand = (OnOffType.OFF.compareTo((OnOffType) command));
+            logger.debug("The command being sent to pin {} for channel:  is {}", channelUID.getAsString(), pin,
+                    sendCommand);
+            sendActuatorCommand(Integer.toString(sendCommand), pin, channelUID);
+        } else if (command instanceof OpenClosedType) {
+            logger.debug("A command was sent to a sensor type so we are ignoring the command");
+        }
+    }
+
+    /**
+     * Process a {@link WebHookEvent} that has been received by the Servlet from a Konnected module with respect to a
+     * sensor event
+     *
+     * @param pin     the pin number which which was activated
+     * @param state   the state of the pin
+     * @param setAuth the token sent with the response
+     */
+    public void handleWebHookEvent(KonnectedModuleGson event, String sentAuth) {
+        // if we receive a command upteate the thing status to being online
+        updateStatus(ThingStatus.ONLINE);
+        // get the zone number based off of the index location of the pin value
+        String sentZone = Integer.toString(Arrays.asList(PIN_TO_ZONE).indexOf(event.getPin()));
+        // check that the zone number is in one of the channelUID definitions
+        getThing().getChannels().forEach(channel -> {
+            ChannelUID channelId = channel.getUID();
+            String zoneNumber = channelId.getId().substring((channelId.getId().length() - 1));
+            // if the string zone that was sent equals the last digit of the channelId found process it as the
+            // channelId else do nothing
+            if (sentZone.equalsIgnoreCase(zoneNumber)) {
+                logger.debug("The channelID of the event is: {}, the Auth Token is: {}", channelId, sentAuth);
+                // check that the auth token sent matches the authToken configured on the Thing
+                if (sentAuth.endsWith(config.authToken)) {
```

Comment:
Does the module really use the auth token for the callbacks as well? I had expected that it uses it only when accepting requests itself?

Reply:
But the access TO the module is done with the same token via pure HTTP - so this can be easily spoofed. So I do not really see much security gain here. But yes, it certainly helps against random url accessed to this endpoint without knowing any token.